src/components/Tabs/ScatterPlotTab.tsx

```diff
@@ -716,15 +740,31 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
       layout.uirevision = uiRevision.current;
     }
 
-    Plotly.react(plotDivRef.current, traces, layout, PLOTLY_CONFIG).then(() => {
-      setIsRendering(false);
-      lastRenderedState.current = {
-        xAxisFeatureKey,
-        yAxisFeatureKey,
-        rangeType,
-        ...props,
-      };
-    });
+    try {
+      Plotly.react(plotDivRef.current, traces, layout, PLOTLY_CONFIG).then(() => {
+        setIsRendering(false);
+        lastRenderedState.current = {
+          xAxisFeatureKey,
+          yAxisFeatureKey,
+          rangeType,
+          ...props,
+        };
+      });
+    } catch (error) {
+      console.error(error);
```

Comment:
could you move your sanitizeNumericDataArrays into this block so you're only cleaning the data if it fails? and then if it fails a second time show the alert?

Reply:
The structure of how the plot is rendered might make that a bit difficult? Currently the data is being filtered by range at the start of `renderPlot`, and there are additional methods that take in the (sanitized) data and perform additional operations on it (for example, highlighting the currently selected object or track). That means that we'd have to perform all of those calculations twice, and I'd have to restructure the method to do the try/catch checking.

Do you think we're losing something from sanitizing the data preemptively?

Reply:
I was just trying to avoid mapping over the data more than is necessary,  I'm assuming these are pretty large arrays

Reply:
That makes sense. I'll add a note to revisit this if it causes performance issues.